Hashing/DOUBLE HASHING/DOUBLEHASHING.cpp

```diff
@@ -0,0 +1,370 @@
+
+        print("The key deleted with value : ");
+        cout << deleteNode(key);
+    }
+
+    int getValueInMap(int key)
+    {
+
+        int hashIndex = (hash1(key) + hash2(key)) % capacity;
+        // finding the key for value
+
+        while (hashNodeArray[hashIndex] != NULL)
+        {
+            if (hashNodeArray[hashIndex] && hashNodeArray[hashIndex]->key == key)
+            {
+                return hashNodeArray[hashIndex]->value;
+            }
+
+            hashIndex++;
+            hashIndex = hashIndex % capacity;
+        }
+        cout << endl
+             << "Key Not found.";
+        return 0;
+    }
+
+    // void get value
+
+    void getValue()
+    {
+        // taking Input from the user
+
+        int key;
+        print("Enter the key to find the value: ");
+        cin >> key;
+
+        print("Value is : ");
+        cout << getValueInMap(key);
+    }
+
+    // Getting a size
+
+    int sizeOfMap()
+    {
+        return size;
+    }
+
+    // void function to get a size so that it does not return something menu can run soomthly
+
+    void getSize()
+    {
+        print("The size of Map is : ");
+        cout << sizeOfMap();
+    }
+
+    void display()
+    {
+        //			cout<<endl<<"Your HashMap -- ";
+        cout << endl;
+        for (int i = 0; i < capacity; i++)
+        {
+            cout << i << " -- ";
+            if (hashNodeArray[i] != NULL)
+            {
+                cout << hashNodeArray[i]->key << " -> " << hashNodeArray[i]->value;
+            }
+            cout << endl;
+        }
+    }
+
+    // function to see collision
+
+    void seeCollision()
+    {
+        cout << this->collisions;
+    }
+};
+
+// Print function
+
+void print(string message)
+{
+    cout << endl
+         << message;
+}
+
+int main()
+{
+    // For Calculating Time
+
+    auto start = high_resolution_clock::now();
+
+    // -- end For Calculating Time
+    int capacity;
+    print("Enter the capacity for which the key will be moded to and array will created on this - ");
+    cin >> capacity;
+    HashMap map(capacity);
+
+    // while for menu --
+    bool menu = true;
+    while (menu)
+    {
+        print("Collisions - ");
+        map.seeCollision();
+        print("Your Hash Map -- ");
+        map.display();
+
+        cout << endl;
+
+        print("Enter 1 for Add Key Value operation");
+        print("Enter 2 for Delete with key operation");
+        print("Enter 3 for Size of Map operation");
+        print("Enter 4 for Get value From Map operation");
+
+        print("Enter 5 to exit");
+        cout << endl;
+        int input;
+        cin >> input;
+        int data;
+
+        switch (input)
+        {
+        case 1:
+            map.insertForMenu();
+            break;
+        case 2:
+            map.deleteNodeVoid();
+            break;
+        case 3:
+            map.getSize();
+            break;
+        case 4:
+            map.getValue();
+            break;
+
```

Comment:
Use "menu = false" in case 5. Program shouldn't be terminate if user enter invalid choice. Use default case for handle this.

Reply:
resolved